enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/RaftMachine.java

```diff
@@ -257,6 +257,10 @@ private boolean leaderChanged( Outcome outcome, MemberId oldLeader )
         {
             return true;
         }
+        else if ( oldLeader != null && outcome.getLeader() == null )
```

Comment:
Wouldn't it be better to simply have it set to null but make the users handle that case?